src/coreclr/jit/morph.cpp

```diff
@@ -3660,7 +3633,7 @@ GenTree* Compiler::fgMorphIndexAddr(GenTreeIndexAddr* indexAddr)
         }
 
         if (((index->gtFlags & (GTF_ASG | GTF_CALL | GTF_GLOB_REF)) != 0) ||
-            gtComplexityExceeds(index, MAX_INDEX_COMPLEXITY) || index->OperIs(GT_LCL_FLD) ||
+            gtComplexityExceeds(index, MAX_ARR_COMPLEXITY) || index->OperIs(GT_LCL_FLD) ||
```

Comment:
This line is a correctness fix. We were checking the index field using the arr complexity.

Reply:
Ok, thanks. We should probably PR things like that separately.

Reply:
Guess it's not actually a functional change so no need for a separate PR
```C++
    const int MAX_ARR_COMPLEXITY   = 4;
    const int MAX_INDEX_COMPLEXITY = 4;
```

Reply:
Yeah I think it's a copy-paste mistake.